pages/search/advanced.tsx

```diff
@@ -0,0 +1,209 @@
+        id="advancedSearchForm"
+        className={styles.advancedSearchForm}
+        method="post"
+        action={`${BASE_URL}/search`}
+        onSubmit={handleSubmit}
+      >
+        <SimpleGrid columns={2} gap="grid.m">
+          <Fieldset id="advancedSearchLeft">
+            {textInputFields.map(({ key, name, label }) => {
+              return (
+                <TextInput
+                  id={key}
+                  labelText={label}
+                  type="text"
+                  name={name}
+                  value=""
+                  key={key}
+                />
+              )
+            })}
+            <Select
+              id="languageSelect"
+              name="language"
+              labelText="Language"
+              aria-labelledby="languageSelect-label"
+            >
+              {languageOptions.map((language) => {
+                return (
+                  <option value={language.value} key={language.value}>
+                    {language.label}
+                  </option>
+                )
+              })}
+            </Select>
+          </Fieldset>
+          <Fieldset id="advancedSearchRight">
+            <Label htmlFor="dates" id="dates-label">
+              Date
+            </Label>
+            <FormRow id="dates" className={styles.dateFields}>
+              <FormField className={styles.formField}>
+                <DatePicker
+                  id="dateAfter"
+                  nameFrom="dateAfter"
+                  dateType="year"
+                  labelText="From"
+                  helperText="e.g. 1901"
+                  initialDate=""
+                />
+              </FormField>
+              <FormField className={styles.formField}>
+                <DatePicker
+                  id="dateBefore"
+                  nameFrom="dateBefore"
+                  dateType="year"
+                  labelText="To"
+                  helperText="e.g. 2000"
+                  initialDate=""
+                />
+              </FormField>
+            </FormRow>
+            <Label htmlFor="formats" id="formats-label">
+              Format
+            </Label>
+            <Fieldset id="formats" className={styles.formatFields}>
+              <FormRow>
+                <FormField className={styles.formField}>
+                  {materialTypeOptions.slice(0, 4).map((materialType) => {
+                    return (
+                      <Checkbox
+                        className={styles.checkbox}
+                        id={materialType.value}
+                        key={materialType.value}
+                        labelText={materialType.label}
+                        name={materialType.value}
+                        value={materialType.value}
+                      />
+                    )
+                  })}
+                </FormField>
+                <FormField className={styles.formField}>
+                  {materialTypeOptions.slice(4).map((materialType) => {
```

Comment:
Reduce magic numbers "4". Perhaps include also an explanation of why only four from PM.